addons/calendar_sms/security/calendar_sms_security.xml

```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo noupdate="1">
```

Comment:
After discussing this with TDE, we came to the conclusion that there is indeed a problem in the record rules of crm_sms, event_sms, ...
Those rules are meant to restrict the access to certain models for create/write/unlink, but not for read.

So, instead of adding a new rule here, we will modify existing rules with:
`<field name="perm_read" eval="False"/>`

So the rule will not apply on the read part and your feature should work properly.
This change has to be done as a FIX in the 13.0 version.

Can you try it out and make a separate branch/PR with it?
Would be nice if you added some unit tests for it that fail without your change and work with it.
Your unit test should ensure that group_user always has access to read the sms.templates, and doesn't have access to creat/write/unlink.

If this is unclear, don't hesitate to ping me on Discord.

Thanks.

Reply:
Here is the PRs for the fix:
COM: https://github.com/odoo/odoo/pull/50446
ENT: https://github.com/odoo/enterprise/pull/10307